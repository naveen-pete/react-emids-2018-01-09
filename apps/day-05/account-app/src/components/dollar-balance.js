import React, { Component } from 'react';
import { connect } from 'react-redux';

import { getConversionRate } from '../actions/currency-converter';

export class DollarBalance extends Component {
  componentDidMount() {
    this.props.getConversionRate();
  }

  computeBalanceInDollars() {
    const { balance, conversionRate } = this.props;

    return conversionRate > 0 ? balance / conversionRate : 0;
  }

  render() {
    return <h3>In USD: {this.computeBalanceInDollars()}</h3>;
  }
}

export default connect(
  ({ balance, conversionRate }) => ({ balance, conversionRate }),
  { getConversionRate }
)(DollarBalance);
