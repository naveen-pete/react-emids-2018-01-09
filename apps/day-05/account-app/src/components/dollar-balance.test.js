import React from 'react';
import { shallow } from 'enzyme';
import { DollarBalance } from './dollar-balance';

describe('DollarBalance', () => {
  const mockGetConversionRate = jest.fn();
  const props = {
    balance: 0,
    conversionRate: 0,
    getConversionRate: mockGetConversionRate
  };
  let dollarBalance = shallow(<DollarBalance {...props} />);

  it('renders properly', () => {
    expect(dollarBalance).toMatchSnapshot();
  });

  it('dispatches the `getConversionRate()` method it receives from props', () => {
    expect(mockGetConversionRate).toHaveBeenCalled();
  });

  describe('when there are valid props', () => {
    beforeEach(() => {
      props.balance = 100;
      props.conversionRate = 10;

      dollarBalance = shallow(<DollarBalance {...props} />);
    });

    it('displays the correct balance in USD', () => {
      expect(dollarBalance.find('h3').text()).toEqual('In USD: 10');
    });
  });
});
