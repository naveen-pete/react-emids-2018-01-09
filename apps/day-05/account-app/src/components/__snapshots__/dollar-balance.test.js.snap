// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DollarBalance renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DollarBalance
    balance={0}
    conversionRate={0}
    getConversionRate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "In USD: ",
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      "In USD: ",
      0,
    ],
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "In USD: ",
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        "In USD: ",
        0,
      ],
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
